app/models/concerns/single_sign_on_concern.rb

```diff
@@ -40,8 +40,10 @@ def get_data_from_sso_mapping(sso_mapping)
     ## @param sso_mapping {String} must be of form 'user._field_' or 'profile._field_'. Eg. 'user.email'
     ## @param data {*} the data to put in the given key. Eg. 'user@example.com'
     def set_data_from_sso_mapping(sso_mapping, data)
+      return if data.nil? || data.blank? || mapped_from_sso&.include?(sso_mapping)
```

Comment:
I'm not sure about the last part of the condition `return if ... || mapped_from_sso&.include?(sso_mapping)` will prevent an update from the SSO, doesn't it?

For exemple, if the field `email` is mapped and provided, when the SSO provided a changed value (the user has changed his email on the mainstream database), this must be refacted in fab-manager